sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -184,8 +227,8 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
         }
     }
 
-    // Aggregation strategy can handle queries with a single distinct group.
-    if (distinctAggGroups.size > 1) {
```

Comment:
For `count(distinct c1), count(c2) FILTER ...`, we don't apply this rule previously and it works well. Why do we need to apply this rule now?

Reply:
Let's use `If (distinctAggGroups.size > 1 || distinctAggs.exists(_.filter.isDefined))` now.